fpdf/image_parsing.py

```diff
@@ -127,24 +132,160 @@ def get_img_info(img, image_filter="AUTO", dims=None):
         ):
             info["smask"] = _to_data(img, image_filter, select_slice=alpha_channel)
 
+    dp = f"/Predictor 15 /Colors {dpn} /Columns {w}"
+    
+    if img.mode == "1":
```

Comment:
Note that a this point the image has already been converted to RGBA, due to lines 78-79 (formerly 73-74)

For context, see https://github.com/PyFPDF/fpdf2/pull/527

Reply:
I'm not sure I follow, line 78 is

```python
if img.mode not in ("1", "L", "LA", "RGB", "RGBA", "P", "PA"):
    img = img.convert("RGBA")
```

so if `img.mode == "1"` then `img.mode not in ("1", "L", "LA", "RGB", "RGBA", "P", "PA")` is false and thus the conversion to RGBA is not called right?

Reply:
My bad, you are totally right ! 🤦‍♂️ Wooops, my bad 😊

However there is already an `if img.mode == "1"` on line 83:
maybe the code you added would be better to replaced after this `if`,
to avoid multiple conditional and group the logic relavant to monochromatic images in a ingle code block?

Reply:
I can reorganize the code but it won't be as nice as you wish I think because the `dp` variable requires the whole block to be done to be calculated (because it uses `dpn` which is computed in each of the blocks). If you still want me to reorganize the code I can, no problem

Reply:
Alright then, leave it as it is!